.github/workflows/build-release.yml

```diff
@@ -0,0 +1,99 @@
+---
+  name: build-release
+  on:
+    push:
+      tags: [ v* ]
+  
+  jobs:
+    lint:
+      name: lint
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v3
+        - uses: actions/setup-go@v4
+          with:
+            go-version: '1.21'
+            cache: false
+        - name: golangci-lint
+          uses: golangci/golangci-lint-action@v3
+          with:
+            version: latest
+            args: --timeout 5m
+    test:
+      name: test
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v3
+        - name: run tests using make
+          run: make test
+    docker:
+      name: docker
+      runs-on: ubuntu-latest
+      needs:
+        - lint
+        - test
+      steps:
+        - uses: actions/checkout@v2
+        - uses: docker/setup-qemu-action@v1
+        - uses: docker/setup-buildx-action@v1
+        - uses: docker/login-action@v1
+          with:
+            registry: ghcr.io
+            username: ${{ github.repository_owner }}
+            password: ${{ secrets.GITHUB_TOKEN }}
+        - uses: docker/metadata-action@v3
+          id: meta
+          with:
+            images: ghcr.io/${{ github.repository }}
+            tags: |
+              type=ref,event=branch
+              type=ref,event=pr
+              type=semver,pattern={{version}}
+              type=semver,pattern={{major}}.{{minor}}
+              type=sha,prefix=
+  
+        - uses: docker/build-push-action@v2
+          with:
+            file: "Dockerfile"
+            context: .
+            platforms: linux/amd64
+            push: true
+            tags: |
+              ${{ steps.meta.outputs.tags }}
+            labels: ${{ steps.meta.outputs.labels }}
+    helm-chart:
+      name: helm-chart
+      needs:
+          - docker
+      runs-on: ubuntu-latest
+      steps:
+        - name: Checkout
+          uses: actions/checkout@v3
+          with:
+            fetch-depth: 0
+  
+        - name: Configure Git
```

Comment:
Please replace with the s3-operator one: https://github.com/snapp-incubator/s3-operator/blob/main/.github/workflows/build-release.yaml#L56-L67